README.md

```diff
@@ -1,602 +1,370 @@
+
+  | Spec type      | Corresponding Rails test class        |
+  | -------------- | --------------------------------      |
+  | [model][]      |                                       |
+  | [controller][] | [`ActionController::TestCase`][]      |
+  | [mailer][]     | `ActionMailer::TestCase`              |
+  | [job][]        |                                       |
+  | [view][]       | `ActionView::TestCase`                |
+  | [routing][]    |                                       |
+  | [helper][]     | `ActionView::TestCase`                |
+  | [request][]    | [`ActionDispatch::IntegrationTest`][] |
+  | [feature][]    |                                       |
+  | [system][]     | [`ActionDispatch::SystemTestCase`][]  |
+
+Follow the links above to see examples of each spec type,
+or for official Rails API documentation on the given `TestCase` class.
+
+> **Note: This is not a checklist.**
+>
+> Ask a hundred developers how to test an application,
+> and you’ll get a hundred different answers.
+>
+> RSpec Rails provides thoughtfully selected features
+> to encourage good testing practices, but there’s no “right” way to do it.
+> Ultimately, it’s up to you to decide how your test suite will be composed.
+
+[request]: https://relishapp.com/rspec/rspec-rails/docs/request-specs/request-spec
+[feature]: https://www.relishapp.com/rspec/rspec-rails/docs/feature-specs/feature-spec
+[system]: https://relishapp.com/rspec/rspec-rails/docs/system-specs/system-spec
+[model]: https://www.relishapp.com/rspec/rspec-rails/docs/model-specs
+[controller]: https://www.relishapp.com/rspec/rspec-rails/docs/controller-specs
+[mailer]: https://relishapp.com/rspec/rspec-rails/docs/mailer-specs
+[job]: https://relishapp.com/rspec/rspec-rails/docs/job-specs/job-spec
+[view]: https://www.relishapp.com/rspec/rspec-rails/docs/view-specs/view-spec
+[routing]: https://www.relishapp.com/rspec/rspec-rails/docs/routing-specs
+[helper]: https://www.relishapp.com/rspec/rspec-rails/docs/helper-specs/helper-spec
+[`ActionDispatch::IntegrationTest`]: https://api.rubyonrails.org/classes/ActionDispatch/IntegrationTest.html
+[`ActionDispatch::SystemTestCase`]: https://api.rubyonrails.org/classes/ActionDispatch/SystemTestCase.html
+[`ActionController::TestCase`]: https://api.rubyonrails.org/classes/ActionController/TestCase.html
+
+### I’m writing a spec file from scratch. How do I assign it a type?
+
+Simply place the spec [in the appropriate folder][]
+(_e.g.,_ `spec/models/` for model specs)
+and RSpec will set its type automatically.
+
+If you want to be extra explicit,
+you can set the `:type` option at the top of the file, like so:
 
 ```ruby
-# rspec-rails-1.x
-assigns[key] = value
+# spec/models/user_spec.rb
 
-# rspec-rails-2.x+
-assign(key, value)
-```
-
-### `rendered`
-
-This represents the rendered view.
-
-```ruby
-render
-expect(rendered).to match /Some text expected to appear on the page/
+RSpec.describe User, type: :model do
```

Comment:
could we bring this example up to above where the table of types is?

Reply:
Like [this](https://github.com/rspec/rspec-rails/pull/2042/files#diff-04c6e90faac2675aa89e2176d2eec7d8R236)?